include/micm/util/cuda_vector_matrix.hpp

```diff
@@ -0,0 +1,65 @@
+#include <type_traits>
+#include <micm/util/vector_matrix.hpp>
+#include <micm/util/cuda_vector_matrix.cuh>
+
+namespace micm {
+
+    template<class T, std::size_t L = DEFAULT_VECTOR_SIZE>
+    class CudaVectorMatrix : public VectorMatrix<T, L> {
+    private:
+        double* d_data_;
+
+    public:
+        CudaVectorMatrix() requires(std::is_same_v<T, double>)
+         : VectorMatrix<T,L>()
+        {
+          micm::cuda::malloc_vector(d_data_, static_cast<unsigned int>(0));
+        }
+        CudaVectorMatrix()
+         : VectorMatrix<T,L>()
+        {}
+
+        CudaVectorMatrix(std::size_t x_dim, std::size_t y_dim) requires(std::is_same_v<T, double>)
+          : VectorMatrix<T,L>(x_dim, y_dim)
+        {
+          micm::cuda::malloc_vector(d_data_, this->data_.size());
+        }
+        CudaVectorMatrix(std::size_t x_dim, std::size_t y_dim)
+          : VectorMatrix<T,L>(x_dim, y_dim)
+        {}
+
+        CudaVectorMatrix(std::size_t x_dim, std::size_t y_dim, T initial_value) requires(std::is_same_v<T, double>)
+          : VectorMatrix<T,L>(x_dim, y_dim, initial_value)
+        {
+          micm::cuda::malloc_vector(d_data_, this->data_.size());
```

Comment:
should `initial_value` be passed as an argument to the CUDA kernel?

Reply:
I believe we are going with the design choice to make any memory movement to the device an explicit call by the user.  So in this case, the user would call `auto matrix = CudaVectorMatrix<...>(...)` and then `matrix.CopyToDevice()` manually.  Which matches what the process will be when we start calling `ForEach(...)` as well.